src/tests/suite/daos_container.c

```diff
@@ -771,6 +771,223 @@ co_set_prop(void **state)
+		assert_int_equal(rc, -DER_NO_PERM);
+
+		print_message("Delete with privs from container ACL only\n");
+
+		cont_acl = daos_acl_dup(cont_prop->dpp_entries[0].dpe_val_ptr);
+		assert_non_null(cont_acl);
+		rc = daos_acl_get_ace_for_principal(cont_acl, DAOS_ACL_OWNER,
+						    NULL,
+						    &update_ace);
+		assert_int_equal(rc, 0);
+		update_ace->dae_allow_perms = DAOS_ACL_PERM_CONT_ALL;
+
+		print_message("- getting container handle\n");
+		rc = daos_cont_open(arg->pool.poh, arg->co_uuid, DAOS_COO_RW,
+				    &coh, NULL, NULL);
+		assert_int_equal(rc, 0);
+
+		print_message("- updating cont ACL to restore delete privs\n");
+		rc = daos_cont_update_acl(coh, cont_acl, NULL);
+		assert_int_equal(rc, 0);
+
+		print_message("- closing container\n");
+		rc = daos_cont_close(coh, NULL);
+		assert_int_equal(rc, 0);
+
+		print_message("Deleting container now should succeed\n");
+		rc = daos_cont_destroy(arg->pool.poh, arg->co_uuid, 1, NULL);
+		assert_int_equal(rc, 0);
+
+		/* Clear cont uuid since we already deleted it */
+		uuid_clear(arg->co_uuid);
+	}
+
+	daos_acl_free(cont_acl);
+	daos_prop_free(pool_prop);
+	daos_prop_free(cont_prop);
+	test_teardown((void **)&arg);
+}
+
+static void
+co_destroy_allowed_by_pool(void **state)
+{
+	test_arg_t	*arg0 = *state;
+	test_arg_t	*arg = NULL;
+	daos_prop_t	*pool_prop;
+	daos_prop_t	*cont_prop;
+	int		 rc;
+
+	rc = test_setup((void **)&arg, SETUP_EQ, arg0->multi_rank,
+			DEFAULT_POOL_SIZE, NULL);
+	assert_int_equal(rc, 0);
+
+	/* pool gives the owner all privs, including delete cont */
+	pool_prop = get_daos_prop_with_owner_acl_perms(DAOS_ACL_PERM_POOL_ALL,
+						       DAOS_PROP_PO_ACL);
+
+	/* container doesn't give delete privs to the owner */
+	cont_prop = get_daos_prop_with_owner_acl_perms(DAOS_ACL_PERM_CONT_ALL &
+						       ~DAOS_ACL_PERM_DEL_CONT,
+						       DAOS_PROP_CO_ACL);
+
+	while (!rc && arg->setup_state != SETUP_CONT_CREATE)
+		rc = test_setup_next_step((void **)&arg, NULL, pool_prop,
+					  cont_prop);
+	assert_int_equal(rc, 0);
+
+	if (arg->myrank == 0) {
+		print_message("Deleting container with only pool-level "
+			      "perms\n");
+		rc = daos_cont_destroy(arg->pool.poh, arg->co_uuid, 1, NULL);
+		assert_int_equal(rc, 0);
+
+		/* Clear cont uuid since we already deleted it */
+		uuid_clear(arg->co_uuid);
+	}
+
+	daos_prop_free(pool_prop);
+	daos_prop_free(cont_prop);
+	test_teardown((void **)&arg);
+}
+
+static void
+expect_cont_open_access(test_arg_t *arg0, uint64_t perms, uint64_t flags,
+			int exp_result)
+{
+	test_arg_t	*arg = NULL;
+	daos_prop_t	*prop;
+	int		 rc;
+
+	rc = test_setup((void **)&arg, SETUP_EQ, arg0->multi_rank,
+			DEFAULT_POOL_SIZE, NULL);
+	assert_int_equal(rc, 0);
+
+	arg->cont_open_flags = flags;
+	prop = get_daos_prop_with_owner_acl_perms(perms,
+						  DAOS_PROP_CO_ACL);
+
+	while (!rc && arg->setup_state != SETUP_CONT_CONNECT)
+		rc = test_setup_next_step((void **)&arg, NULL, NULL, prop);
+
+	if (arg->myrank == 0) {
```

Comment:
may not be worth removing the check at this point but if I understand correctly all ranks will have the rc value (that rank 0 got when it opened the container - since it got broadcast to all ranks). So all MPI ranks may be able to run the asserts here unconditionally. Yes OK I see in tests/suite/daos_pool.c expect_pool_connect_access() it does not protect the assert_int_equal() calls.

Reply:
Ah, got it. Thanks. I'll remove the check if there are any further comments that need to be addressed.